perception/sub8_perception/src/sub8_perception/start_gate.cc

```diff
@@ -0,0 +1,191 @@
+            //     for (int i = 0; i < gate_line_buffer[0].size(); i++) {
+            // cv::Point2i s = gate_line_buffer[0][i][1] - gate_line_buffer[0][i][0];
+            // cv::Point2i center = cv::Point2i(s.x / 2 + s.x, s.y / 2 + s.y);
+            // std::cout << center << std::endl;
+
+            //         //Make bigger buffer. After like 30 calculate mean and average. Outlier stuff.
+            //     }
+            // }
+            for (size_t i = 0; i < gate_line_buffer[29].size(); i++) {
+                cv::Point2i s = gate_line_buffer[29][i][1] - gate_line_buffer[29][i][0];
+                cv::Point2i center = cv::Point2i(s.x / 2 + s.x, s.y / 2 + s.y);
+                // std::cout << center << std::endl;
+                if (abs(center.x - mean(accX)) > sqrt(variance(accX)) ||
+                        abs(center.y - mean(accY) > sqrt(variance(accY)))) {
+                    gate_line_buffer[29].erase(gate_line_buffer[29].begin() + i);
+                    i--;
+                    std::cout << "Deleted: " << center << std::endl;
+                    continue;
+                }
+                if (abs(s.x - mean(accSizeX)) > sqrt(variance(accSizeX)) ||
+                        abs(s.y - mean(accSizeY)) > sqrt(variance(accSizeY))) {
+                    gate_line_buffer[29].erase(gate_line_buffer[29].begin() + i);
+                    i--;
+                    std::cout << "Deleted size: " << s << std::endl;
+                    continue;
+                }
+            }
+            cv::Mat show = cv::Mat::zeros(rows, cols, CV_8UC3);
+
+            for (size_t i = 0; i < gate_line_buffer[29].size(); i++) {
+                // for (int j = 0; j < gate_line_buffer[0][i].size(); j++) {
+                // cv::line(show, gate_line_buffer[0][i][j], gate_line_buffer[0][i][(j + 1) % 4], cv::Scalar(0, 0, 255), 1, 8 );
+                // }
+                cv::rectangle(show, gate_line_buffer[29][i][0], gate_line_buffer[29][i][1], cv::Scalar(255, 0, 0), 2, 8, 0);
+            }
+            cv::imshow("ay", show);
+            cv::waitKey(30);
+        }
+        findGate(image_msg);
+
+    }
+}
+
+
+void Sub8StartGateDetector::findGate(const sensor_msgs::ImageConstPtr &image_msg) {
+
+    cv::Mat cvImageMat = cv_bridge::toCvShare(image_msg, "bgr8")->image;
+    cv::imshow("show", cvImageMat);
+    cols = cvImageMat.cols;
+    rows = cvImageMat.rows;
+    cv::Scalar mean;
+    cv::Scalar stddev;
+    cv::meanStdDev(cvImageMat, mean, stddev);
+    // cv::imshow("view", cvImageMat);
+
+    cv::Mat output;
+    cv::inRange(cvImageMat, mean - 2 * stddev, mean + 2 * stddev, output);
+    // cv::imshow("d", output);
+    cv::bitwise_not(output, output);
+    cv::dilate(output, output, cv::Mat(), cv::Point(-1, -1), 2, 1, 1);
+
+    cv::Mat skel(output.size(), CV_8UC1, cv::Scalar(0));
+    cv::Mat temp(output.size(), CV_8UC1);
+
+    cv::Mat element = cv::getStructuringElement(cv::MORPH_CROSS, cv::Size(3, 3));
+
+    bool done;
```

Comment:
`done` has an undefined value, as does the variable `max` below. You should always initialize your variables when you declare them unless you're sure that you won't be generating some undefined behavior.